src/publisher.ts

```diff
@@ -75,10 +86,26 @@ export class Publisher {
   }
 
   private sayHello() {
+    const promises: Promise<void>[] = [];
+
     for (let iface of this.getInterfaces()) {
       const message = new Buffer(this.buildMessage(iface.address));
-      this.client.send(message, 0, message.length, PORT, iface.broadcast);
+      promises.push(new Promise<void>((resolve, reject) => {
+        if (!this.client) {
+          return reject(new Error('no client'));
+        }
+
+        this.client.send(message, 0, message.length, PORT, iface.broadcast, err => {
+          if (err) {
+            reject(err);
+          } else {
+            resolve();
+          }
+        });
+      }));
     }
+
+    return Promise.all(promises);
```

Comment:
Since sayHello is a private function it might make more sense to call a error handler, so the CLI could subscribe to the "runtime" errors:
```
const publisher = new Publisher(...
publisher.onError((err)=> this.env.log.error(err));
```
^ something like that